angularSTEP/src/app/discover-display/discover-display.component.ts

```diff
@@ -0,0 +1,207 @@
+import {Component, OnInit, Input, SimpleChanges} from '@angular/core';
+import {
+  AngularFirestore,
+  AngularFirestoreCollection,
+} from '@angular/fire/firestore';
+import {Router} from '@angular/router'
+import {User} from '../user';
+import {Converter} from '../converter';
+import {Recipe} from '../recipe';
+import {RecipeConverter} from '../recipe-converter'
+
+@Component({
+  selector: 'app-discover-display',
+  templateUrl: './discover-display.component.html',
+  styleUrls: ['./discover-display.component.scss'],
+})
+export class DiscoverDisplayComponent implements OnInit {
+  recipeCollection: AngularFirestoreCollection<Recipe> = this.afs.collection(
+    'recipes'
+  );
+  recipes: firebase.firestore.QueryDocumentSnapshot<Recipe>[];
+
+  userCollection: AngularFirestoreCollection<User> = this.afs.collection(
+    'users'
+  );
+  users: firebase.firestore.QueryDocumentSnapshot<User>[];
+  @Input() sortType: string;
+  @Input() contentType: string;
+  @Input() direction: string;
+
+  constructor(private afs: AngularFirestore, private router: Router) {}
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes && changes.sortType) {
+      this.sortType = changes.sortType.currentValue;
+    }
+    if (changes && changes.contentType) {
+      this.contentType = changes.contentType.currentValue;
+    }
+    if (changes && changes.direction) {
+      this.direction = changes.direction.currentValue;
+    }
+    this.getCollectionData();
+  }
+
+  ngOnInit() {}
+
+  getCollectionData() {
+    if (this.sortType && this.contentType && this.direction !== undefined) {
+      switch (this.contentType) {
+        case 'Recipes': {
+          this.getRecipes();
+          break;
+        }
+        case 'Users': {
+          this.getUsers();
+          break;
+        }
+        default: {
+          this.getRecipes();
+        }
+      }
+    }
+  }
+
+  getRecipes() {
```

Comment:
It looks like you could have a switch statement for setting a variable called orderByField, and not repeat the logic in each case.